File: icure-sdk/src/commonMain/kotlin/com/icure/sdk/crypto/entities/EntityWithEncryptionMetadataTypeName.kt
package com.icure.sdk.crypto.entities

import com.icure.sdk.model.Article

/**
 * Types of encryptable entities, to support segregation of secure delegation keys.
 * These types do not include stubs and metadata entities (such as SecureDelegationKeyMap) since they use the actual
 * type of the related entity, and does not include services as they use the type of Contact.
 */
enum class EntityWithEncryptionMetadataTypeName(
	/**
	 * Uniquely identifies the entity type, it is included as a "salt" in the secure delegation key, to segregate the
	 * secure delegation key spaces of different entity types.
	 * We must never change these values since it could break the secure delegation key derivation, making some data
	 * inaccessible to anonymous users.
	 */
	val id: String
) {
	Article("Article"),
	AccessLog("AccessLog"),
	CalendarItem("CalendarItem"),
	Classification("Classification"),
	ClassificationTemplate("ClassificationTemplate"),
	Contact("Contact"),
	Document("Document"),
	Form("Form"),
	HealthElement("HealthElement"),
	Invoice("Invoice"),
	MaintenanceTask("MaintenanceTask"),
	Message("Message"),
	Patient("Patient"),
	Receipt("Receipt"),
	TimeTable("TimeTable"),
	Topic("Topic")
}

File: python-wrapper/src/python/crypto/entities/EntityWithEncryptionMetadataTypeName.py
from typing import Union, Dict
from enum import Enum

class EntityWithEncryptionMetadataTypeName(Enum):
	Article = "Article"
	AccessLog = "AccessLog"
	CalendarItem = "CalendarItem"
	Classification = "Classification"
	ClassificationTemplate = "ClassificationTemplate"
	Contact = "Contact"
	Service = "Service"
	Document = "Document"
	Form = "Form"
	HealthElement = "HealthElement"
	Invoice = "Invoice"
	MaintenanceTask = "MaintenanceTask"
	Message = "Message"
	Patient = "Patient"
	Receipt = "Receipt"
	TimeTable = "TimeTable"
	Topic = "Topic"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'EntityWithEncryptionMetadataTypeName':
		if data == "Article":
			return EntityWithEncryptionMetadataTypeName.Article
		elif data == "AccessLog":
			return EntityWithEncryptionMetadataTypeName.AccessLog
		elif data == "CalendarItem":
			return EntityWithEncryptionMetadataTypeName.CalendarItem
		elif data == "Classification":
			return EntityWithEncryptionMetadataTypeName.Classification
		elif data == "ClassificationTemplate":
			return EntityWithEncryptionMetadataTypeName.ClassificationTemplate
		elif data == "Contact":
			return EntityWithEncryptionMetadataTypeName.Contact
		elif data == "Service":
			return EntityWithEncryptionMetadataTypeName.Service
		elif data == "Document":
			return EntityWithEncryptionMetadataTypeName.Document
		elif data == "Form":
			return EntityWithEncryptionMetadataTypeName.Form
		elif data == "HealthElement":
			return EntityWithEncryptionMetadataTypeName.HealthElement
		elif data == "Invoice":
			return EntityWithEncryptionMetadataTypeName.Invoice
		elif data == "MaintenanceTask":
			return EntityWithEncryptionMetadataTypeName.MaintenanceTask
		elif data == "Message":
			return EntityWithEncryptionMetadataTypeName.Message
		elif data == "Patient":
			return EntityWithEncryptionMetadataTypeName.Patient
		elif data == "Receipt":
			return EntityWithEncryptionMetadataTypeName.Receipt
		elif data == "TimeTable":
			return EntityWithEncryptionMetadataTypeName.TimeTable
		elif data == "Topic":
			return EntityWithEncryptionMetadataTypeName.Topic
		else:
			raise Exception(f"{data} is not a valid value for EntityWithEncryptionMetadataTypeName enum.")

__all__ = ['EntityWithEncryptionMetadataTypeName']